src/addons/matchrecursive.js

```diff
@@ -180,9 +195,23 @@ export default (XRegExp) => {
                     }
                 }
             } else {
-                const delimSide = rightMatch ? 'right' : 'left';
-                const errorPos = rightMatch ? delimStart : outerStart;
-                throw new Error(`Unbalanced ${delimSide} delimiter found in string at position ${errorPos}`);
+                // eslint-disable-next-line no-lonely-if
+                if (unbalancedDelimiters === 'error') {
+                    const delimSide = rightMatch ? 'right' : 'left';
+                    const errorPos = rightMatch ? delimStart : outerStart;
+                    throw new Error(`Unbalanced ${delimSide} delimiter found in string at position ${errorPos}`);
+                } else if (unbalancedDelimiters.startsWith('skip')) {
+                    if (rightMatch) {
+                        rightMatch = null;
+                    } else {
+                        delimEnd = unbalancedDelimiters === 'skip-lazy' ?
+                            1 :
+                            (XRegExp.exec(str, left, outerStart)[0].length + outerStart) || 1;
```

Comment:
Needs to be `(XRegExp.exec(str, left, outerStart)[0].length || 1) + outerStart`, otherwise it won't work correctly if `outerStart > 0`. Will fix in a follow up.

Reply:
agh yea you're right, my bad!